src/redisjson.rs

```diff
@@ -46,23 +46,33 @@ impl Format {
 /// Backwards compatibility convertor for RedisJSON 1.x clients
 ///
 pub struct Path {
-    pub path: String,
-    pub fixed: String,
+    original_path: String,
+    fixed_path: Option<String>,
```

Comment:
Just a thought,
Can we even consider (also later) avoiding the `fixed_path` member and calculate it when calling `get_path` which will return the moved String? (either original or fixed if needed)
(assuming that once we pass it to JSONPath selector we don't need it anymore)
This will shorten the lifetime of the cloned/mutated strings to the minimum, but it will probably have none to very small impact (unless `paths: &mut Vec<Path>` vector in method `to_json` is big)